gpu/src/cuda/shaders/utils.h

```diff
@@ -0,0 +1,17 @@
+#ifndef UTILS_H
+#define UTILS_H
+
+/// Reverses the `log2(size)` first bits of `i`
+__device__ unsigned reverse_index(unsigned i, unsigned size)
+{
+    if (size == 1)
+    { // TODO: replace this statement with an alternative solution.
+        return i;
+    }
+    else
+    {
+        return __brev(i) >> (__clz(size) + 1);
+    }
```

Comment:
If the branching is the issue, you can use the multiplication trick:
```suggestion
    return (size == 1) * i | (size != 1) * (__brev(i) >> (__clz(size) + 1));
```